static/app/components/notificationActions/forms/pagerdutyForm.tsx

```diff
@@ -0,0 +1,149 @@
+import {useState} from 'react';
+import styled from '@emotion/styled';
+
+import {Button} from 'sentry/components/button';
+import {DropdownMenu, MenuItemProps} from 'sentry/components/dropdownMenu';
+import {
+  NotificationActionButtonContainer,
+  NotificationActionCell,
+  NotificationActionFormContainer,
+} from 'sentry/components/notificationActions/notificationActionItem';
+import {IconChevron} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import space from 'sentry/styles/space';
+import {
+  AvailableNotificationAction,
+  NotificationAction,
+} from 'sentry/types/notificationActions';
+
+interface PagerdutyFormProps {
+  /**
+   * The notification action being represented
+   */
+  action: Partial<NotificationAction>;
+  onCancel: () => void;
+  onChange: (name: string, value: any) => void;
+  onSave: () => void;
+  /**
+   * Map of pagerduty integration IDs to available actions for those IDs
+   */
+  pagerdutyIntegrations: Record<number, AvailableNotificationAction[]>;
+}
+
+function PagerdutyForm({
+  action,
+  onCancel,
+  onChange,
+  onSave,
+  pagerdutyIntegrations,
+}: PagerdutyFormProps) {
+  const [selectedAccount, setSelectedAccount] = useState(
+    action.integrationId
+      ? pagerdutyIntegrations[action.integrationId][0].action.integrationName
+      : ''
+  );
+  const [selectedDisplay, setSelectedDisplay] = useState(action.targetDisplay ?? '');
+
+  const accountOptions: MenuItemProps[] = [];
+  Object.keys(pagerdutyIntegrations).forEach(integrationId => {
+    // Get the name of the integration for the integrationId from the first
+    // AvailableNotificationAction element in the array
+    const integrationName =
+      pagerdutyIntegrations[integrationId][0].action.integrationName;
+    accountOptions.push({
+      key: integrationName,
+      label: integrationName,
+      onAction: value => {
+        onChange('integrationId', integrationId);
+        setSelectedAccount(value);
+      },
+    } as MenuItemProps);
+  });
```

Comment:
can avoid casting and pushing
```suggestion
  const accountOptions = Object.keys(pagerdutyIntegrations).map<MenuItemProps>(integrationId => {
    // Get the name of the integration for the integrationId from the first
    // AvailableNotificationAction element in the array
    const integrationName =
      pagerdutyIntegrations[integrationId][0].action.integrationName;
    accountOptions.push({
      key: integrationName,
      label: integrationName,
      onAction: value => {
        onChange('integrationId', integrationId);
        setSelectedAccount(value);
      },
    });
  });
```